File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <vector>
#include <iostream>
#include <stack>

using namespace std;

/*
stack에 데이터를 넣기
stack의 top이 집어넣으려는 값이랑 같으면 continue;
*/
vector<int> solution(vector<int> arr) 
{
    
    std::ios::sync_with_stdio(0);
    std::cin.tie(0);
    
    vector<int> answer;

    stack<int> stack;
    for(int x : arr){
        if(!stack.empty() && stack.top() == x)continue;
        stack.push(x);
       answer.push_back(x);
    }
    

    return answer;
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;
import java.io.*;
/*
중복제거
문제
set에 넣어서 제거해버리자
*/

public class Solution {
    public List<Integer> solution(int []arr) {
        List<Integer> answer = new ArrayList<>();
        
        Stack<Integer> stack = new Stack<>();
        for(int x : arr){
            if(!stack.empty() && stack.peek()==x)continue;
            stack.push(x);
        }
        
        for(int z : stack){
            answer.add(z);
        }
        // [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.

        return answer;
    }
}